### DISCUSSION TITLE
Timeline.Item 组件 `color` 调整

### DISCUSSION THREAD

**Discussion by Wxh16144** [ant-design/ant-design#40064] [Ideas]
## 描述

将 color 默认四个表示状态属性改为预设状态，颜色支持预设颜色值

## 动机

现目前用的四个预设关键词 `blue`、`red`、`green`、`gray` ,  [这个示例文档](https://ant.design/components/timeline-cn#components-timeline-demo-color) 中描述

 > 绿色用于已完成、成功状态，红色表示告警或错误状态，蓝色可表示正在进行或其他默认状态，灰色表示未完成或失效状态。

可以参考 [`<Badge />` 组件](https://ant.design/components/badge-cn#components-badge-demo-status)，将原有的四个属性改为预设状态关键词

## 实现

1. 将 `blue` 改为 `processing` , `red` 改为 `error`，`green` 改为 `success`, `gray` 改为 `default`，另外一个预设状态 `warning` 作为新增
2. 将预设颜色添加到 color 属性中支持。

## 缺点

1. 对于原先的 color 属性来说只是一个破坏性变更

**Comment by afc163** [comment]
变成新增就好，老的废弃 + wanring。

**Reply by Wxh16144** [reply]
> 变成新增就好，老的废弃 + wanring。

但是如果想实现第二点，原来是关键词和 `presetColors`
https://github.com/ant-design/ant-design/blob/b7abcc60ac9da0327474c5cbeccdd18a2fcda6e0/components/theme/interface/presetColors.ts 冲突了，有没有必要实现第二点呢

**Reply by MadCcc** [reply]
对于用户来说 blue 也不等同于 primary，其实更偏向一个 bug 修复，不过从新增 Color 和 Status 统一层面来讲更适合作为 feature。

**Reply by Wxh16144** [reply]
如果作为 bug 修复的话，应该不仅仅是修复 blue 这个关键词，而是将 presetColors 处理进去。

如果作为 feature 的话，原先使用的人将会受到影响。

有更多的讨论么～
